src/lib.rs

```diff
@@ -527,45 +527,75 @@ pub async fn get_assets_vault(
 
 #[cfg(not(target_arch = "wasm32"))]
 pub async fn transfer_assets(transfers: TransfersRequest) -> Result<TransfersSerializeResponse> {
-    use strict_encoding::strict_serialize;
+    use crate::data::structs::{BlindedOrNotOutpoint, ChangeTansfer, DeclareRequest};
 
     let resp = transfer_asset(transfers).await?;
 
     let psbt = serialize_psbt(&resp.psbt);
     let psbt = base64::encode(&psbt);
     let disclosure = serde_json::to_string(&resp.disclosure)?;
-    let transfers = resp
-        .transfers
-        .into_iter()
-        .map(|(c, seals)| {
-            let consignment = strict_serialize(&c).expect("Consignment information must be valid");
-            let consignment = util::bech32m_zip_encode("rgbc", &consignment)
-                .expect("Strict encoded information must be a valid consignment");
-
-            FinalizeTransfer {
-                consignment,
-                beneficiaries: seals.into_iter().map(|s| s.to_string()).collect(),
-            }
-        })
-        .collect();
+
+    let mut transfers = vec![];
+    let mut change_transfers = vec![];
+    for (index, asset_transfer_info) in resp.transaction_info.iter().enumerate() {
+        transfers.push(FinalizeTransfer {
+            consignment: asset_transfer_info.consignment.clone(),
```

Comment:
Hi @josediegorobles,

I think this line has incorrect, because the `transaction_info` does not contains the transfer state, but consignment contract.

I don't known if you need the original consignment information in this part, but before the change, we return the transfer state.

Because of this, the tests broke when try check the accept transfer.